crates/sp-lightclient/src/lib.rs

```diff
@@ -0,0 +1,308 @@
+// Copyright (C) 2021 Subspace Labs, Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Light client substrate primitives for Subspace.
+#![forbid(unsafe_code)]
+#![warn(rust_2018_idioms, missing_docs)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use sp_api::HeaderT;
+use sp_consensus_subspace::digests::{
+    find_global_randomness_descriptor, find_pre_digest, find_salt_descriptor,
+    find_solution_range_descriptor, CompatibleDigestItem, Error as DigestError,
+    GlobalRandomnessDescriptor, PreDigest, SaltDescriptor, SolutionRangeDescriptor,
+};
+use sp_consensus_subspace::FarmerPublicKey;
+use sp_std::cmp::Ordering;
+use subspace_core_primitives::{
+    BlockWeight, PublicKey, Randomness, RecordSize, RewardSignature, Salt, SegmentSize,
+    SolutionRange,
+};
+use subspace_solving::{derive_global_challenge, derive_target, REWARD_SIGNING_CONTEXT};
+use subspace_verification::{check_reward_signature, verify_solution, VerifySolutionParams};
+
+#[cfg(test)]
+mod tests;
+
+/// HeaderExt describes an extended block chain header at a specific height along with some computed values.
+pub struct HeaderExt<Header> {
+    /// Actual header of the subspace block chain at a specific height
+    pub header: Header,
+    /// Global randomness after importing the header above.
+    /// This is same as the parent block unless update interval is met.
+    pub derived_global_randomness: Randomness,
+    /// Solution range after importing the header above.
+    /// This is same as the parent block unless update interval is met.
+    pub derived_solution_range: SolutionRange,
+    /// Salt after importing the header above.
+    /// This is same as the parent block unless update interval is met.
+    pub derived_salt: Salt,
```

Comment:
It doesn't seem like any of these need to exist here, you can always derive them with a method call on Header instead.